src/views/Home/Home.jsx

```diff
@@ -17,13 +31,33 @@ const Home = ({ token, setToken }) => {
     setToken(newToken);
   };
 
+  const handleJoinList = async () => {
+    //first we query the firebase database with the input token
+    const q = query(collection(db, joinList));
+    //then we take a snapshot of the results by calling getDocs() on our query
+    const querySnapshot = await getDocs(q);
+    // console.log('QUERY SNAPSHOT', querySnapshot.size)
+    // if the snapshot.size is greater than one, we can assume it's a valid list, set the token in state, and navigate the user to the list view
+    if (querySnapshot.size > 1) {
+      localStorage.setItem('token', joinList);
+      setToken(joinList);
+      navigate(`/list`);
+    }
+    // otherwise set an error in state we can pass to the join list component
+    setFormError(
+      'This token does not match an existing shopping list. Please check your input and try again.',
+    );
+  };
```

Comment:
🎉 I know this was a hurdle to get working! Great job! I think your error message is descriptive and helpful – it shows empathy for your user!